Lambda Function Unable To Find Header Files

Context: 
I'm quite new to AWS and am go illiterate as well (for now). My current aim is to reproduce the experiment from [this blog](https://blog.nelhage.com/post/building-llvm-in-90s/) so that I can see how llama works and then hand it off to an intern to do more research to understand how we can integrate llama into our current meson-and-ninja based build system to build our firmware applications. However, in my efforts to build LLVM, I hit the following type of errors:
`[98/2572] Building CXX object lib/Support/CMakeFiles/LLVMSupport.dir/AMDGPUMetadata.cpp.o
FAILED: lib/Support/CMakeFiles/LLVMSupport.dir/AMDGPUMetadata.cpp.o 
/home/njpau/go/bin/llamac++  -DGTEST_HAS_RTTI=0 -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -Ilib/Support -I/home/njpau/git/llvm-project/llvm/lib/Support -I/usr/include/libxml2 -Iinclude -I/home/njpau/git/llvm-project/llvm/include -fPIC -fvisibility-inlines-hidden -Werror=date-time -std=c++11 -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wno-missing-field-initializers -pedantic -Wno-long-long -Wno-maybe-uninitialized -Wdelete-non-virtual-dtor -Wno-comment -fdiagnostics-color -ffunction-sections -fdata-sections -O0    -fno-exceptions -fno-rtti -MD -MT lib/Support/CMakeFiles/LLVMSupport.dir/AMDGPUMetadata.cpp.o -MF lib/Support/CMakeFiles/LLVMSupport.dir/AMDGPUMetadata.cpp.o.d -o lib/Support/CMakeFiles/LLVMSupport.dir/AMDGPUMetadata.cpp.o -c /home/njpau/git/llvm-project/llvm/lib/Support/AMDGPUMetadata.cpp
_root/home/njpau/git/llvm-project/llvm/lib/Support/AMDGPUMetadata.cpp:16:28: fatal error: llvm/ADT/Twine.h: No such file or directory
 #include "llvm/ADT/Twine.h"
                            ^
compilation terminated.
Running llamacc: invoke: exit 1
`

Everything I have done today is from scratch, including setting up my free tier AWS account. These are the steps I've gone through:
1. Installed go on my 4C/8T Debian x86_64 GNU/Linux machine and set up an AWS account for myself.
2. Installed llama from source using the instructions in the Install llama section under Getting Started in the README. Did the llamac++ symlinking as well.
3. Installed awscli2 and my creds have been appropriately configured and stored in ~/.aws
4. Ran `llama bootstrap`. After completion, I went to my aws console and verified that the cloudformation stack had been successful created and that I had an empty S3 bucket, and empty ECR repo and an IAM role for basic lambda execution.
5. Ran `home/njpau/go/src/github.com/nelhage/llama/scripts/build-gcc-image --local-headers` to build my debian image and lambda function. Verified that I see 'gcc' image and function in my aws console under ECR and Lambda. 
5. Cloned the [llvm-project](https://github.com/llvm/llvm-project) from github and checked-out the release/8.x branch (because I'm running python3.5 on my system)
6. `cd llvm-project && mkdir -p build && cd build`
7. Ran `cmake -GNinja \
  -DCMAKE_BUILD_TYPE=Release \
  -DLLVM_ENABLE_PROJECTS=clang \
  -DLLVM_TARGETS_TO_BUILD=X86 \
  -DLLVM_PARALLEL_LINK_JOBS=8 \
  -DLLVM_BUILD_TOOLS=OFF \
  -DLLVM_BUILD_UTILS=OFF  \
  -DCMAKE_CXX_FLAGS_RELEASE="-O0" \
  -DCLANG_ENABLE_STATIC_ANALYZER=OFF  \
  -DCLANG_ENABLE_ARCMT=OFF \
   ../llvm/` to test building locally first.
8. Ran `ninja build`. Build successfully completed in roughly 20 mins.
9. Deleted the build dir, created a new one and cd-ed into it.
10. Ran `LLAMACC_LOCAL=1 CC=llamacc CXX=llamac++ \
  cmake -GNinja \
  -DCMAKE_BUILD_TYPE=Release \
  -DLLVM_ENABLE_PROJECTS=clang \
  -DLLVM_TARGETS_TO_BUILD=X86 \
  -DLLVM_PARALLEL_LINK_JOBS=8 \
  -DLLVM_BUILD_TOOLS=OFF \
  -DLLVM_BUILD_UTILS=OFF  \
  -DCMAKE_CXX_FLAGS_RELEASE="-O0" \
  -DCLANG_ENABLE_STATIC_ANALYZER=OFF  \
  -DCLANG_ENABLE_ARCMT=OFF \
   ../llvm/`
11. Ran `ninja build -j100`. And that's when the build bombed out with 99 errors similar to the one above (i.e., reporting 'No such file or directory' for required headers). 

What am I doing wrong here? Have I configured llama incorrectly? I have tried to follow the README + [this blog post](https://blog.nelhage.com/post/building-llvm-in-90s/).

`njpau@debian-vm-1:~/git/llvm-project/build$ LLAMACC_VERBOSE=1 ninja -v lib/Support/CMakeFiles/LLVMSupport.dir/AMDGPUMetadata.cpp.o 
[1/1] /home/njpaul/go/bin/llamac++  -DGTEST_HAS_RTTI=0 -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -Ilib/Support -I/home/njpau/git/llvm-project/llvm/lib/Support -I/usr/include/libxml2 -Iinclude -I/home/njpau/git/llvm-project/llvm/include -fPIC -fvisibility-inlines-hidden -Werror=date-time -std=c++11 -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wno-missing-field-initializers -pedantic -Wno-long-long -Wno-maybe-uninitialized -Wdelete-non-virtual-dtor -Wno-comment -fdiagnostics-color -ffunction-sections -fdata-sections -O0    -fno-exceptions -fno-rtti -MD -MT lib/Support/CMakeFiles/LLVMSupport.dir/AMDGPUMetadata.cpp.o -MF lib/Support/CMakeFiles/LLVMSupport.dir/AMDGPUMetadata.cpp.o.d -o lib/Support/CMakeFiles/LLVMSupport.dir/AMDGPUMetadata.cpp.o -c /home/njpau/git/llvm-project/llvm/lib/Support/AMDGPUMetadata.cpp
FAILED: lib/Support/CMakeFiles/LLVMSupport.dir/AMDGPUMetadata.cpp.o 
/home/njpau/go/bin/llamac++  -DGTEST_HAS_RTTI=0 -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -Ilib/Support -I/home/njpau/git/llvm-project/llvm/lib/Support -I/usr/include/libxml2 -Iinclude -I/home/njpau/git/llvm-project/llvm/include -fPIC -fvisibility-inlines-hidden -Werror=date-time -std=c++11 -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wno-missing-field-initializers -pedantic -Wno-long-long -Wno-maybe-uninitialized -Wdelete-non-virtual-dtor -Wno-comment -fdiagnostics-color -ffunction-sections -fdata-sections -O0    -fno-exceptions -fno-rtti -MD -MT lib/Support/CMakeFiles/LLVMSupport.dir/AMDGPUMetadata.cpp.o -MF lib/Support/CMakeFiles/LLVMSupport.dir/AMDGPUMetadata.cpp.o.d -o lib/Support/CMakeFiles/LLVMSupport.dir/AMDGPUMetadata.cpp.o -c /home/njpau/git/llvm-project/llvm/lib/Support/AMDGPUMetadata.cpp
2021/07/13 13:13:35 run cpp -MM: ["c++" "-fPIC" "-fvisibility-inlines-hidden" "-Werror=date-time" "-std=c++11" "-Wall" "-Wextra" "-Wno-unused-parameter" "-Wwrite-strings" "-Wcast-qual" "-Wno-missing-field-initializers" "-pedantic" "-Wno-long-long" "-Wno-maybe-uninitialized" "-Wdelete-non-virtual-dtor" "-Wno-comment" "-fdiagnostics-color" "-ffunction-sections" "-fdata-sections" "-O0" "-fno-exceptions" "-fno-rtti" "-D" "GTEST_HAS_RTTI=0" "-D" "_GNU_SOURCE" "-D" "__STDC_CONSTANT_MACROS" "-D" "__STDC_FORMAT_MACROS" "-D" "__STDC_LIMIT_MACROS" "-I" "lib/Support" "-I" "/home/njpau/git/llvm-project/llvm/lib/Support" "-I" "/usr/include/libxml2" "-I" "include" "-I" "/home/njpau/git/llvm-project/llvm/include" "-M" "-MF" "-" "/home/njpau/git/llvm-project/llvm/lib/Support/AMDGPUMetadata.cpp"]
2021/07/13 13:13:35 [llamacc] compiling remotely: daemon.InvokeWithFilesArgs{Trace:(*tracing.Propagation)(nil), Function:"gcc", ReturnLogs:false, Args:[]string{"c++", "-I", "_root/home/njpau/git/llvm-project/build", "-I", "_root/home/njpau/git/llvm-project/build/lib/Support", "-I", "_root/home/njpau/git/llvm-project/llvm/lib/Support", "-I", "_root/usr/include/libxml2", "-I", "_root/home/njpau/git/llvm-project/build/include", "-I", "_root/home/njpau/git/llvm-project/llvm/include", "-D", "GTEST_HAS_RTTI=0", "-D", "_GNU_SOURCE", "-D", "__STDC_CONSTANT_MACROS", "-D", "__STDC_FORMAT_MACROS", "-D", "__STDC_LIMIT_MACROS", "-c", "-o", "_root/home/njpau/git/llvm-project/build/lib/Support/CMakeFiles/LLVMSupport.dir/AMDGPUMetadata.cpp.o", "_root/home/njpau/git/llvm-project/llvm/lib/Support/AMDGPUMetadata.cpp", "-MD", "-MF", "_root/home/njpau/git/llvm-project/build/lib/Support/CMakeFiles/LLVMSupport.dir/AMDGPUMetadata.cpp.o.d.tmp", "-fPIC", "-fvisibility-inlines-hidden", "-Werror=date-time", "-std=c++11", "-Wall", "-Wextra", "-Wno-unused-parameter", "-Wwrite-strings", "-Wcast-qual", "-Wno-missing-field-initializers", "-pedantic", "-Wno-long-long", "-Wno-maybe-uninitialized", "-Wdelete-non-virtual-dtor", "-Wno-comment", "-fdiagnostics-color", "-ffunction-sections", "-fdata-sections", "-O0", "-fno-exceptions", "-fno-rtti"}, Stdin:[]uint8(nil), Files:files.List{files.Mapped{Local:files.LocalFile{Path:"/home/njpau/git/llvm-project/llvm/lib/Support/AMDGPUMetadata.cpp", Bytes:[]uint8(nil), Mode:0x0}, Remote:"_root/home/njpau/git/llvm-project/llvm/lib/Support/AMDGPUMetadata.cpp"}}, Outputs:files.List{files.Mapped{Local:files.LocalFile{Path:"/home/njpau/git/llvm-project/build/lib/Support/CMakeFiles/LLVMSupport.dir/AMDGPUMetadata.cpp.o", Bytes:[]uint8(nil), Mode:0x0}, Remote:"_root/home/njpau/git/llvm-project/build/lib/Support/CMakeFiles/LLVMSupport.dir/AMDGPUMetadata.cpp.o"}, files.Mapped{Local:files.LocalFile{Path:"/home/njpau/git/llvm-project/build/lib/Support/CMakeFiles/LLVMSupport.dir/AMDGPUMetadata.cpp.o.d.tmp", Bytes:[]uint8(nil), Mode:0x0}, Remote:"_root/home/njpau/git/llvm-project/build/lib/Support/CMakeFiles/LLVMSupport.dir/AMDGPUMetadata.cpp.o.d.tmp"}}, DropSemaphore:true}
_root/home/njpau/git/llvm-project/llvm/lib/Support/AMDGPUMetadata.cpp:16:28: fatal error: llvm/ADT/Twine.h: No such file or directory
 #include "llvm/ADT/Twine.h"
                            ^
compilation terminated.
Running llamacc: invoke: exit 1
ninja: build stopped: subcommand failed.
`

--- a/cmd/llamacc/dependencies.go
+++ b/cmd/llamacc/dependencies.go
@@ -46,7 +46,9 @@ func detectDependencies(ctx context.Context, client *daemon.Client, cfg *Config,
 		preprocessor.Args = append(preprocessor.Args, opt.Opt)
 		preprocessor.Args = append(preprocessor.Args, opt.Path)
 	}
-	preprocessor.Args = append(preprocessor.Args, "-M", "-MF", "-", comp.Input)
+	// Use /dev/stdout instead of - because old GCC verions don't
+	// understand `-`. See #50
+	preprocessor.Args = append(preprocessor.Args, "-M", "-MF", "/dev/stdout", comp.Input)
 	var deps bytes.Buffer
 	preprocessor.Stdout = &deps
 	preprocessor.Stderr = os.Stderr

